webhook/resources/engineimage/mutator.go

```diff
@@ -0,0 +1,73 @@
+package engineimage
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/longhorn/longhorn-manager/datastore"
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
+	"github.com/longhorn/longhorn-manager/types"
+	"github.com/longhorn/longhorn-manager/webhook/admission"
+	werror "github.com/longhorn/longhorn-manager/webhook/error"
+	"github.com/pkg/errors"
+)
+
+type engineImageMutator struct {
+	admission.DefaultMutator
+	ds *datastore.DataStore
+}
+
+func NewMutator(ds *datastore.DataStore) admission.Mutator {
+	return &engineImageMutator{ds: ds}
+}
+
+func (e *engineImageMutator) Resource() admission.Resource {
+	return admission.Resource{
+		Name:       "engineimages",
+		Scope:      admissionregv1.NamespacedScope,
+		APIGroup:   longhorn.SchemeGroupVersion.Group,
+		APIVersion: longhorn.SchemeGroupVersion.Version,
+		ObjectType: &longhorn.EngineImage{},
+		OperationTypes: []admissionregv1.OperationType{
+			admissionregv1.Create,
```

Comment:
Basically, nothing can be moved to `engineImage.Update` or `engineImage.UpdateStatus`..

Reply:
Okay, I was thinking we could mutate the `spec.image`, but this would cause the EngineImage CR name to be changed.

Reply:
Gotcha. Normalizing the `spec.image` and mutating it is good idea, but the previous logic does not include it. Thus, I did't do it to avoid unexpected side effects.